UT4MasterServer/Controllers/UnrealTournamentMatchmakingController.cs

```diff
@@ -235,8 +235,29 @@ public async Task<IActionResult> UpdateGameServerPlayers(string id, [FromBody] G
 		if (server.OwningClientID != user.Session.ClientID)
 			Unauthorized();
 
-		server.PublicPlayers = serverOnlyWithPlayers.PublicPlayers;
-		server.PrivatePlayers = serverOnlyWithPlayers.PrivatePlayers;
+		// handle player list update
+		foreach (var player in serverOnlyWithPlayers.PublicPlayers)
+		{
+			if (!server.PublicPlayers.Where(x => x == player).Any())
+			{
+				server.PublicPlayers.Add(player);
+			}
+			if (server.PrivatePlayers.Where(x => x == player).Any())
+			{
+				server.PrivatePlayers.Remove(player);
+			}
+		}
+		foreach (var player in serverOnlyWithPlayers.PrivatePlayers)
+		{
+			if (!server.PrivatePlayers.Where(x => x == player).Any())
+			{
+				server.PrivatePlayers.Add(player);
+			}
+			if (server.PublicPlayers.Where(x => x == player).Any())

```

Comment:
else

Reply:
same as above, but now reverse role of `PrivatePlayers` and `PublicPlayers`